posts/Martin/2015-07-13-introduction-to-hibernate-search-genericjpa.adoc

```diff
@@ -0,0 +1,460 @@
+and improves it by adding features like clustering and support for
+mappings of Java Beans.
+
+== Adding Search to our Book store
+
+Now, let’s take a look at how this is done for our Book store. 
+First, we need to annotate our JPA entity with some extra information.
+
+[source,java]
+----
+package org.hibernate.search.genericjpa.test.entities;
+ 
+import org.hibernate.search.annotations.DocumentId;
+import org.hibernate.search.annotations.Field;
+import org.hibernate.search.annotations.Indexed;
+import org.hibernate.search.genericjpa.annotations.InIndex;
+ 
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Table;
+ 
+@Entity
+@Table(name = "Book")
+@Indexed
+@InIndex
+public class Book {
+ 
+	@Id
+	@DocumentId
+	@Field
+	@Column(name = "name")
+	private String name;
+ 
+	@Field
+	@Column(name = "author")
+	private String author;
+ 
+	public String getName() {
+		return name;
+	}
+ 
+	public void setName(String name) {
+		this.name = name;
+	}
+ 
+	public String getAuthor() {
+		return author;
+	}
+ 
+	public void setAuthor(String author) {
+		this.author = author;
+	}
+}
+----
+
+=== What did we do?
+- Class-Level:
+	* added @Indexed annotation
+		** This is needed for Hibernate Search to recognize this entity-class as a Top-Level index class.
+	* added @InIndex annotation
+		** This is a special annotation needed by Hibernate Search GenericJPA on every 
+			entity that somehow is found in any index. Just put it there and you’ll be fine
+- Field-Level:
+	* added @DocumentId/@Field on name
+		** Hibernate Search needs to know which field is used to identify this entity in the index.
+			this produces a field with the name “id” in the index.
+			We also want to store the name into a field called “name”.
+	* added @Field on author
+		** apart from searching for the name we also want to be able to search for
+			the author of the book. this is stored into the field called “author” in the index.
+			
+== Starting up the engine
+
+As Hibernate Search GenericJPA is not integrated into Hibernate ORM we have to 
+manually start everything up. But this is not hard, at all:
+
+[source,java]
+----
+//first, let's start up the basic JPA persistence.
+EntityManagerFactory emf = 
+  Persistence.createEntityManagerFactory( "EclipseLink_MySQL" );
+EntityManager em = emf.createEntityManager();
+ 
+//now, let's get the Properties
+Properties hsearchConfiguration = "...";
+//start our JPASearchFactoryController
+JPASearchFactoryController searchController =
+  Setup.createSearchFactory( emf, hsearchConfiguration );
+----
+
+You may have noticed loading a properties file in this snippet.
+This is the configuration needed for Hibernate Search. Let’s take a look at it next.
+
+[source]
+----
+# we use are using MySQL. this is needed to 
+# create the triggers for updating
+org.hibernate.search.genericjpa.searchfactory.triggerSource=\
```

Comment:
You should describe the general architectural approach using triggers somewhere. I only vaguely remember it, other readers won't have a clue at all of what's going on ;)